lib/finch/http1/conn.ex

```diff
@@ -110,28 +110,35 @@ defmodule Finch.Conn do
 
     start_time = Telemetry.start(:request, metadata, extra_measurements)
 
-    case HTTP1.request(conn.mint, req.method, full_path, req.headers, stream_or_body(req.body)) do
-      {:ok, mint, ref} ->
-        case maybe_stream_request_body(mint, ref, req.body, receive_timeout) do
-          {:ok, mint} ->
-            Telemetry.stop(:request, start_time, metadata, extra_measurements)
-            start_time = Telemetry.start(:response, metadata, extra_measurements)
-            response = receive_response([], acc, fun, mint, ref, receive_timeout)
-            handle_response(response, conn, metadata, start_time, extra_measurements)
-
-          {:error, mint, error} ->
-            handle_request_error(
-              conn,
-              mint,
-              error,
-              metadata,
-              start_time,
-              extra_measurements
-            )
-        end
-
-      {:error, mint, error} ->
-        handle_request_error(conn, mint, error, metadata, start_time, extra_measurements)
+    try do
+      case HTTP1.request(conn.mint, req.method, full_path, req.headers, stream_or_body(req.body)) do
+        {:ok, mint, ref} ->
+          case maybe_stream_request_body(mint, ref, req.body, receive_timeout) do
+            {:ok, mint} ->
+              Telemetry.stop(:request, start_time, metadata, extra_measurements)
+              start_time = Telemetry.start(:response, metadata, extra_measurements)
+              response = receive_response([], acc, fun, mint, ref, receive_timeout)
+              handle_response(response, conn, metadata, start_time, extra_measurements)
+
+            {:error, mint, error} ->
+              handle_request_error(
+                conn,
+                mint,
+                error,
+                metadata,
+                start_time,
+                extra_measurements
+              )
+          end
+
+        {:error, mint, error} ->
+          handle_request_error(conn, mint, error, metadata, start_time, extra_measurements)
+      end
```

Comment:
I think @keathley is not a big fan of `with` statement, but it fells right in this specific case _imho_ to prevent the `handle_request_error/6` repetition:

```suggestion
      with {:ok, mint, ref} <- HTTP1.request(conn.mint, req.method, full_path, req.headers, stream_or_body(req.body)),
           {:ok, mint} <- maybe_stream_request_body(mint, ref, req.body, receive_timeout) do
        Telemetry.stop(:request, start_time, metadata, extra_measurements)
        start_time = Telemetry.start(:response, metadata, extra_measurements)
        response = receive_response([], acc, fun, mint, ref, receive_timeout)
        handle_response(response, conn, metadata, start_time, extra_measurements)
      else
        {:error, mint, error} ->
          handle_request_error(conn, mint, error, metadata, start_time, extra_measurements)
      end
```

Reply:
I want to minimize changes that are not related to the fix itself so I am leaving the existing code alone. It only shows up in the diff because of the indentation change.